src/bonefish/messages/wamp_message.hpp

```diff
@@ -28,22 +29,47 @@ class wamp_message
 {
 public:
     wamp_message();
-    virtual ~wamp_message();
+    virtual ~wamp_message() = default;
 
-    wamp_message(const wamp_message& other) = delete;
+    wamp_message(const wamp_message&) = delete;
+    wamp_message(wamp_message&&) = delete;
     wamp_message& operator=(wamp_message const&) = delete;
+    wamp_message& operator=(wamp_message&&) = delete;
+
+    std::unique_ptr<msgpack::zone>&& release_zone();
```

Comment:
Ok cool. I originally wanted to avoid unique_ptr and just use zone's built in move semantics however, msgpack::unpacker stores its zone as a unique_ptr so unfortunately we are stuck with unique_ptr for the time being.